src/app/modules/OpenshiftStreams/dialogs/CreateInstance/hooks/useCreateInstance.ts

```diff
@@ -46,24 +46,43 @@ export const useCreateInstance =
         if (isServiceApiError(error)) {
           const { code } = error?.response?.data || {};
 
-          switch (code) {
```

Comment:
Errors to consider:

Wrong Instance Type/Size:
https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L92

Invalid Region:
https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L65

Invalid Provider:
https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L62

Trial: You have trial already
https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L56

Terms have changed:
https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L41

Max Trial allowed reached:
https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L20

Reply:
CLI for reference: https://github.com/redhat-developer/app-services-cli/blob/main/pkg/cmd/kafka/create/create.go#L252-L264

Reply:
> Wrong Instance Type/Size: https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L92

This one and error 42 too are about providing an invalid combination of plan and size id, eg `standard.x1` is valid `foo.bar` isn't. Since users can't type that string manually, these errors aren't relevant for the UI.

> 
> Invalid Region: https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L65
> 
> Invalid Provider: https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L62

As above, since the user can select only valid values, these errors aren't really relevant to the UI.

> 
> Trial: You have trial already https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L56

This is not specific to trial instances, it's for when the cluster is out of capacity for reasons not related to the user.
It's handled for standard instances, where for trial we show the generic trial unavailable error.

> 
> Terms have changed: https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L41

This should be an app-services-ui concern I think, not specific to this dialog, and I think it's handled somehow (but not sure).

> 
> Max Trial allowed reached: https://github.com/redhat-developer/app-services-sdk-js/blob/main/packages/kafka-management-sdk/src/errors.ts#L20

That's covered when you open the dialog. We don't have a specific error message when there was no trial in use by the user when opening the dialog and then the trial happens to be used when creating the dialog. We'll show the generic error for trial instances in this case.

Reply:
Fully agree. It was more or less informational